errors.go

```diff
@@ -134,6 +130,32 @@ func (e *pgconnError) Unwrap() error {
 	return e.err
 }
 
+// ErrTimeout occurs when an error was caused by a timeout. Specifically, it wraps an error which is
+// context.Canceled, context.DeadlineExceeded, or an implementer of net.Error where Timeout() is true.
+type ErrTimeout struct {
```

Comment:
I think this type should be private. This will keep this new type as an implementation detail for `pgconn.Timeout` rather than adding a second way to check if an error is a timeout

Reply:
The intent behind this being public is to allow external code to check for the error directly using the standard unwrapping pattern, like so:

```go
func fetchAndLogOnTimeoutErr(ctx context.Context, sql string) ([]*Result, error) {
	// ...
	res, err := myConn.Exec(ctx, sql).ReadAll()
	var timeoutErr *pgconn.ErrTimeout
	if errors.As(err, &timeoutErr) {
		fmt.Println("Hey, a timeout error!")
	}
	// ...
}
```

Although this is more verbose than calling `pgconn.Timeout`, the transparency it provides is worth the extra line to me as a user of `pgconn`. When using `pgconn`, I'd like to have the option to choose either pattern when checking for timeout errors. If `pgconn.ErrTimeout` is private, the caller is forced to use `pgconn.Timeout`.

If you disagree and would like to keep the error type private, I'll go ahead and make the change. Just wanted to throw in my 2 cents!

Reply:
Let's make it private for now. I think we should make it public in some manner. But I'm thinking that we may want to do a something more comprehensive to identify all errors returned from pgconn. Not sure if exposing the type directly is best until I'm more sure what the long term direction will be. See https://github.com/jackc/pgconn/issues/81#issuecomment-874235445 for the direction I'm thinking.

Reply:
Sure thing, changed: dd6b5b7